packages/assets-controllers/src/NftController.test.ts

```diff
@@ -862,19 +978,33 @@ describe('NftController', () => {
 
       const callActionSpy = jest
         .spyOn(messenger, 'call')
+        // 1. `AccountsController:getAccount`
         .mockReturnValueOnce(OWNER_ACCOUNT)
+        // 2. `AssetsContractController:getERC721OwnerOf`
+        .mockRejectedValueOnce(new Error('Not an ERC721 contract'))
+        // 3. `AssetsContractController:getERC1155BalanceOf`
+        .mockResolvedValueOnce(new BN(1))
+        // 4. `AssetsContractController:getERC721TokenURI`
+        .mockRejectedValueOnce(new Error('Not an ERC721 contract'))
+        // 5. `AssetsContractController:getERC1155TokenURI`
+        .mockResolvedValueOnce('https://testtokenuri.com')
+        // 6. `ApprovalController:addRequest`
         .mockResolvedValueOnce({})
-        .mockReturnValueOnce(OWNER_ACCOUNT);
+        // 7. `AccountsController:getAccount`
+        .mockReturnValueOnce(OWNER_ACCOUNT)
+        // 8. `AssetsContractController:getERC1155AssetName`
+        .mockResolvedValueOnce('testERC1155Name')
+        // 9. `AssetsContractController:getERC1155AssetSymbol`
+        .mockResolvedValueOnce('testERC1155Symbol');
 
       await nftController.watchNft(
         ERC1155_NFT,
         ERC1155,
         'https://etherscan.io',
       );
-      // First call is getInternalAccount. Second call is the approval request.
-      expect(callActionSpy).toHaveBeenCalledTimes(3);
+      expect(callActionSpy).toHaveBeenCalledTimes(9);
       expect(callActionSpy).toHaveBeenNthCalledWith(
-        2,
+        6,
         'ApprovalController:addRequest',
```

Comment:
This is the flow of NftController method calls and messenger call invocations triggered by a `watchNft` call on a ERC1155 asset.

```md
- `#getAddressOrSelectedAddress`
  1. `AccountsController:getAccount`
- `#validateWatchNft` > `isNftOwner`
  2. `AssetsContractController:getERC721OwnerOf` (rejects)
  3. `AssetsContractController:getERC1155BalanceOf`
- `#getNftInformation` > `#getNftInformationFromTokenURI` > `#getNftURIAndStandard`
  4. `AssetsContractController:getERC721TokenURI` (rejects)
  5. `AssetsContractController:getERC1155TokenURI`
- `_requestApproval`
  6. `ApprovalController:addRequest`
- `addNft` > `#getAddressOrSelectedAddress`
  7. `AccountsController:getAccount`
- `#addNftContract` > `#getNftContractInformation` > `#getNftContractInformationFromContract`
  8. `AssetsContractController:getERC1155AssetName`
  9. `AssetsContractController:getERC1155AssetSymbol`
```